File: BOJ/1697.py
# 숨바꼭질

"""
수빈 N
동생 K
이동 X-1, X+1, X*2
"""

from collections import deque

N, K = map(int, input().split())
dist = [-1 for _ in range(100001)]
dist[N] = 0
dq = deque()
dq.append(N)

while dist[K] == -1:
    current = dq.popleft()
    _next = [current - 1, current + 1, current * 2]
    for i in _next:
        if i < 0 or i >= len(dist):
            continue
        if dist[i] != -1:
            continue
        dist[i] = dist[current] + 1
        dq.append(i)

print(dist[K])

File: BOJ/silver/1697.js
// 1697 - 숨바꼭질

const fs = require("fs");
const file = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const [N, K] = fs.readFileSync(file).toString().trim().split(" ").map(Number);

const solution = (N, K) => {
  const dist = Array(100001).fill(-1);
  const queue = [N];
  dist[N] = 0;

  while (dist[K] === -1) {
    const cur = queue.shift();

    for (const next of [cur - 1, cur + 1, cur * 2]) {
      if (next < 0 || next > 100000) continue;
      if (dist[next] !== -1) continue;

      queue.push(next);
      dist[next] = dist[cur] + 1;
    }
  }

  return dist[K];
};

console.log(solution(N, K));
